### PR TITLE
Added the reviews Controller

### PR DESCRIPTION
**Pull request by sarawq**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -1,10 +1,15 @@
 class ReviewsController < ApplicationController
+
   def new
+    @review = Review.new
   end
 
   def create
+    @review = Review.new(review_params)
```
Comment (Isabellerousseau) [inline app/controllers/reviews_controller.rb]: need to define review_params in private method

Comment (sarawq) [reply app/controllers/reviews_controller.rb]: nice

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -1,10 +1,15 @@
 class ReviewsController < ApplicationController
+
   def new
+    @review = Review.new
```
Comment (Isabellerousseau) [inline app/controllers/reviews_controller.rb]: product_reservations POST   /products/:product_id/reservations(.:format)                                             reservations#create
  new_product_reservation GET    /products/:product_id/reservations/new(.:format)                                         reservations#new
          product_reviews POST   /products/:product_id/reviews(.:format)                                                  reviews#create
       new_product_review GET    /products/:product_id/reviews/new(.:format)                                              reviews#new

these are the routes... so we need to call the products or reservations somewhere!!! not sure if these routes are correct though so maybe raise a ticket?

Comment (Isabellerousseau) [reply app/controllers/reviews_controller.rb]: i personally think we have to have another look at the migrations I am afraid as a product has a review and a user has a review but a reservations does not have a review as you are linking back to a particular product or user...

Comment (Isabellerousseau) [reply app/controllers/reviews_controller.rb]: def new
    @product = Product.find(params[:product_id])
    @review = Review.new
  end

  def create
  @product = Product.find(params[:product_id])
  @review = Review.new(review_params)
  @review.product = @product
  if @review.save
      redirect_to product_path(@product)
    else
      render :new
    end
  end

Comment (sarawq) [reply app/controllers/reviews_controller.rb]: The reservation has a review through product that's how I see it, but I will ask

Comment (sarawq) [reply app/controllers/reviews_controller.rb]: Let me know guys if there is anything to change/add

**Pull request merged**